src/components/ClusterNodes/types.ts

```diff
@@ -149,3 +149,52 @@ export const TEXT_COLOR_CLASS = {
     'Not ready': 'cr-5',
 }
 
+interface NodeDataPropType {
+    nodeData: NodeDetail
+    getNodeListData: () => void
+}
+
+export interface NodeActionsMenuProps extends NodeDataPropType {
+    openTerminal: (clusterData: NodeDetail) => void
+}
+
+export interface CordonNodeModalProps extends NodeDataPropType {
+    toggleShowCordonNodeDialog: () => void
+}
+
+export interface DrainNodeModalProps extends NodeDataPropType {
+    toggleShowDrainNodeDialog: () => void
+}
+
+export interface DeleteNodeModalProps extends NodeDataPropType {
+    toggleShowDeleteNodeDialog: () => void
+}
+
+interface NodeActionRequest {
+    clusterId: number
+    name: string
+    version: string
+    kind: string
+}
+
+export interface NodeDeleteRequest extends NodeActionRequest {}
```

Comment:
We can use NodeActionRequest directly.

Reply:
Done. Actually, I kept NodeDeleteRequest intentionally for any additional keys.